### PR TITLE
fix memory leak in wrapper field #1852

### PR DESCRIPTION
**Pull request by kangpinghuang**
Use unique_ptr + new to replace arena to allocate and manage memory for string type fields.

### CODE REVIEW

File: be/src/olap/wrapper_field.h [hunk 1]
```diff
@@ -112,7 +113,10 @@ class WrapperField {
     //include fixed and variable length and null bytes
     size_t _length;
     size_t _var_length;
+    // used for copy
     Arena _arena;
+    // memory for string type field
+    std::unique_ptr<char> _string_content;
```
Comment (imay) [inline be/src/olap/wrapper_field.h]:
```suggestion
    std::unique_ptr<char[]> _string_content;
```

File: be/src/olap/wrapper_field.h [hunk 2]
```diff
@@ -112,7 +113,10 @@ class WrapperField {
     //include fixed and variable length and null bytes
     size_t _length;
     size_t _var_length;
+    // used for copy
     Arena _arena;
```
Comment (imay) [inline be/src/olap/wrapper_field.h]: Remove this?

**Review by imay — CHANGES_REQUESTED** [review]

File: be/src/olap/wrapper_field.h (hunk 1 above)
Comment (kangpinghuang) [reply be/src/olap/wrapper_field.h]: ok

File: be/src/olap/wrapper_field.h (hunk 2 above)
Comment (kangpinghuang) [reply be/src/olap/wrapper_field.h]: ok

File: be/src/olap/wrapper_field.h [hunk 3]
```diff
@@ -99,10 +101,20 @@ class WrapperField {
...
-        _rep->deep_copy_content((char*)cell_ptr(), value, &_arena);
+        if (_is_string_type) {
+            const Slice* src = reinterpret_cast<const Slice*>(value);
+            if (src->size > _var_length) {
+                _string_content.reset(new char[src->size]);
+                _var_length = src->size;
+                Slice* slice = reinterpret_cast<Slice*>(cell_ptr());
+                slice->size = _var_length;
+                slice->data = _string_content.get();
+                memset(slice->data, 0, slice->size);
```
Comment (imay) [inline be/src/olap/wrapper_field.h]: Why memset?

**Review by imay — APPROVED** [review]
LGTM

**Pull request merged**